### PR TITLE
Move UnsupportedOSPlatform from class to methods for MD5 and HMACMD5

### LINKED ISSUE #68556: Consider moving UnsupportedOSPlatform on MD5 from the class to methods
Currently, the whole class of MD5 is marked as `UnsupportedOSPlatform` on Browser:

https://github.com/dotnet/runtime/blob/2ea67f1f42e8170676c9e9591a974e39513390db/src/libraries/System.Security.Cryptography/src/System/Security/Cryptography/MD5.cs#L20-L21

I would propose we move this to individual methods (`Create` and `HashData`), similar to what was done with `RC2`, `DSA`.

The reason I suggest this is because first, someone could technically derive `MD5` and perhaps that implementation works in Browser. The other reason is because there are now helpful constants on the class that have no bearing on a functioning MD5 implementation.

# API Proposal

```diff
namespace System.Security.Cryptography
{
-   [System.Runtime.Versioning.UnsupportedOSPlatformAttribute("browser")]
    public partial class HMACMD5 : System.Security.Cryptography.HMAC
    {
        public const int HashSizeInBits = 128;
        public const int HashSizeInBytes = 16;
+       [System.Runtime.Versioning.UnsupportedOSPlatformAttribute("browser")]
        public HMACMD5() { }
+       [System.Runtime.Versioning.UnsupportedOSPlatformAttribute("browser")]
        public HMACMD5(byte[] key) { }
        public override byte[] Key { get { throw null; } set { } }
        protected override void Dispose(bool disposing) { }
        protected override void HashCore(byte[] rgb, int ib, int cb) { }
        protected override void HashCore(System.ReadOnlySpan<byte> source) { }
+       [System.Runtime.Versioning.UnsupportedOSPlatformAttribute("browser")]
        public static byte[] HashData(byte[] key, byte[] source) { throw null; }
+       [System.Runtime.Versioning.UnsupportedOSPlatformAttribute("browser")]
        public static byte[] HashData(byte[] key, System.IO.Stream source) { throw null; }
+       [System.Runtime.Versioning.UnsupportedOSPlatformAttribute("browser")]
        public static byte[] HashData(System.ReadOnlySpan<byte> key, System.IO.Stream source) { throw null; }
+       [System.Runtime.Versioning.UnsupportedOSPlatformAttribute("browser")]
        public static int HashData(System.ReadOnlySpan<byte> key, System.IO.Stream source, System.Span<byte> destination) { throw null; }
+       [System.Runtime.Versioning.UnsupportedOSPlatformAttribute("browser")]
        public static byte[] HashData(System.ReadOnlySpan<byte> key, System.ReadOnlySpan<byte> source) { throw null; }
+       [System.Runtime.Versioning.UnsupportedOSPlatformAttribute("browser")]
        public static int HashData(System.ReadOnlySpan<byte> key, System.ReadOnlySpan<byte> source, System.Span<byte> destination) { throw null; }
+       [System.Runtime.Versioning.UnsupportedOSPlatformAttribute("browser")]
        public static System.Threading.Tasks.ValueTask<byte[]> HashDataAsync(byte[] key, System.IO.Stream source, System.Threading.CancellationToken cancellationToken = default(System.Threading.CancellationToken)) { throw null; }
+       [System.Runtime.Versioning.UnsupportedOSPlatformAttribute("browser")]
        public static System.Threading.Tasks.ValueTask<int> HashDataAsync(System.ReadOnlyMemory<byte> key, System.IO.Stream source, System.Memory<byte> destination, System.Threading.CancellationToken cancellationToken = default(System.Threading.CancellationToken)) { throw null; }
+       [System.Runtime.Versioning.UnsupportedOSPlatformAttribute("browser")]
        public static System.Threading.Tasks.ValueTask<byte[]> HashDataAsync(System.ReadOnlyMemory<byte> key, System.IO.Stream source, System.Threading.CancellationToken cancellationToken = default(System.Threading.CancellationToken)) { throw null; }
        protected override byte[] HashFinal() { throw null; }
        public override void Initialize() { }
+       [System.Runtime.Versioning.UnsupportedOSPlatformAttribute("browser")]
        public static bool TryHashData(System.ReadOnlySpan<byte> key, System.ReadOnlySpan<byte> source, System.Span<byte> destination, out int bytesWritten) { throw null; }
        protected override bool TryHashFinal(System.Span<byte> destination, out int bytesWritten) { throw null; }
    }

-   [System.Runtime.Versioning.UnsupportedOSPlatformAttribute("browser")]
    public abstract partial class MD5 : System.Security.Cryptography.HashAlgorithm
    {
        public const int HashSizeInBits = 128;
        public const int HashSizeInBytes = 16;
        protected MD5() { }
+       [System.Runtime.Versioning.UnsupportedOSPlatformAttribute("browser")]
        public static new System.Security.Cryptography.MD5 Create() { throw null; }
        [System.Diagnostics.CodeAnalysis.RequiresUnreferencedCodeAttribute("The default algorithm implementations might be removed, use strong type references like 'RSA.Create()' instead.")]
        public static new System.Security.Cryptography.MD5? Create(string algName) { throw null; }
+       [System.Runtime.Versioning.UnsupportedOSPlatformAttribute("browser")]
        public static byte[] HashData(byte[] source) { throw null; }
+       [System.Runtime.Versioning.UnsupportedOSPlatformAttribute("browser")]
        public static byte[] HashData(System.IO.Stream source) { throw null; }
+       [System.Runtime.Versioning.UnsupportedOSPlatformAttribute("browser")]
        public static int HashData(System.IO.Stream source, System.Span<byte> destination) { throw null; }
+       [System.Runtime.Versioning.UnsupportedOSPlatformAttribute("browser")]
        public static byte[] HashData(System.ReadOnlySpan<byte> source) { throw null; }
+       [System.Runtime.Versioning.UnsupportedOSPlatformAttribute("browser")]
        public static int HashData(System.ReadOnlySpan<byte> source, System.Span<byte> destination) { throw null; }
+       [System.Runtime.Versioning.UnsupportedOSPlatformAttribute("browser")]
        public static System.Threading.Tasks.ValueTask<int> HashDataAsync(System.IO.Stream source, System.Memory<byte> destination, System.Threading.CancellationToken cancellationToken = default(System.Threading.CancellationToken)) { throw null; }
+       [System.Runtime.Versioning.UnsupportedOSPlatformAttribute("browser")]
        public static System.Threading.Tasks.ValueTask<byte[]> HashDataAsync(System.IO.Stream source, System.Threading.CancellationToken cancellationToken = default(System.Threading.CancellationToken)) { throw null; }
+       [System.Runtime.Versioning.UnsupportedOSPlatformAttribute("browser")]
        public static bool TryHashData(System.ReadOnlySpan<byte> source, System.Span<byte> destination, out int bytesWritten) { throw null; }
    }
```

### PR DESCRIPTION
**Pull request by vcsjones**
Closes #68556

### CODE REVIEW

File: src/libraries/apicompat/ApiCompatBaseline.NetCoreAppLatestStable.txt [hunk 1]
```diff
@@ -1,4 +1,6 @@
 Compat issues with assembly mscorlib:
+CannotChangeAttribute : Attribute 'System.Runtime.Versioning.UnsupportedOSPlatformAttribute' on 'System.IO.FileStream.Lock(System.Int64, System.Int64)' changed from '[UnsupportedOSPlatformAttribute("ios")]' in the contract to '[UnsupportedOSPlatformAttribute("freebsd")]' in the implementation.
```
Comment (vcsjones) [inline src/libraries/apicompat/ApiCompatBaseline.NetCoreAppLatestStable.txt:2]: I just ran `dotnet build ./src/libraries/apicompat/ApiCompat.proj --no-incremental /p:UpdateBaselineFiles=true` to update this. Most of the unrelated changes appear to just be re-orderings.

File: src/libraries/System.Security.Cryptography/src/System/Security/Cryptography/MD5.cs [hunk 2]
```diff
@@ -35,8 +34,10 @@ protected MD5()
             HashSizeValue = HashSizeInBits;
         }
 
+        [UnsupportedOSPlatform("browser")]
         public static new MD5 Create() => new Implementation();
 
+        [UnsupportedOSPlatform("browser")]
```
Comment (bartonjs) [inline src/libraries/System.Security.Cryptography/src/System/Security/Cryptography/MD5.cs:40]: To match (e.g.) RSA and DSA, this one (the newly Obsoleted `Create(string)`) should be left out (I had skipped it in the proposal).

**Review by bartonjs — APPROVED** [review]

**Review by stephentoub — APPROVED** [review]
